core/requirements.lua

```diff
@@ -107,6 +107,31 @@ end
 
 function Item:IsMet() return ns.PlayerHasItem(self.id, self.count) end
 
+-------------------------------------------------------------------------------
+--------------------------------- PROFESSION ----------------------------------
+-------------------------------------------------------------------------------
+
+local Profession = Class('Profession', Requirement)
+
+function Profession:Initialize(profession)
+    self.profession = strlower(profession):gsub('^%l', string.upper)
+    self.text = self.profession
+end
+
+function Profession:IsMet()
+    local prof1, prof2, archaeology, fishing, cooking, firstAid =
+        GetProfessions()
+    local professions = {prof1, prof2, archaeology, fishing, cooking, firstAid}
+    for i = 1, #professions do
+        if professions[i] ~= nil then
+            if self.profession == GetProfessionInfo(professions[i]) then
```

Comment:
GetProfessionInfo(id) returns at index 1 the translated profession name.
![](http://puu.sh/Jg8xn/ed1e313dd7.jpg)
The condition will only be true for english players.

Reply:
Perhaps it might be better to use the index 2 value which is the icon file. It's not exactly as intuitive but I imagine should be cross-locale. Something like `ns.requirement.Profession(441139)` or to be explicit `ns.requirement.Profession({icon = 441139})` I am surprised there isn't a way to accurately get this information already.

Reply:
Good catch. I didn't notice that index 1 was localized. I do see a table in wow.tools that shows how to match enum types with skills. Maybe I need to just include that instead. https://wow.tools/dbc/?dbc=profession&build=10.0.0.44999#page=1

Reply:
I just pushed a update for this.